drivers/gpio/gpio_sam.c

```diff
@@ -40,9 +40,15 @@ static int gpio_sam_port_configure(const struct device *dev, uint32_t mask,
 	const struct gpio_sam_config * const cfg = dev->config;
 	Pio * const pio = cfg->regs;
 
-	if (flags & GPIO_SINGLE_ENDED) {
-		/* TODO: Add support for Open Source, Open Drain mode */
+	if ((flags & GPIO_OPEN_DRAIN) == GPIO_OPEN_DRAIN) {
+		/* Enable open-drain drive mode */
+		pio->PIO_MDER = mask;
+	} else if ((flags & GPIO_SINGLE_ENDED) != 0) {
+		/* Open-drain is the only supported single-ended mode */
 		return -ENOTSUP;
+	} else {
+		/* Disable open-drain drive mode */
+		pio->PIO_MDDR = mask;
 	}
```

Comment:
NAK on this, you must check `GPIO_SINGLE_ENDED` to enable/disable the feature.

Reply:
This should now be resolved, and I've also rebased onto current (3.2.99) main branch just in case that helps.

Reply:
Perfect! Thank you!